src/Workspaces/Core/Portable/Editing/ImportAdder.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Collections;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editing
+{
+    public abstract class ImportAdder : ILanguageService
+    {
+        /// <summary>
+        /// Adds namespace imports / using directives for namespace references found in the document.
+        /// The <see cref="SyntaxNode"/> with namespace references are annotated for simplification.
```

Comment:
Just so I understand the comment, this means that once imports are added, the ones that imports are added for are marked so the Simplifier can deal with them downstream? A few more words might help here.